devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,359 @@
+        /// </summary>
+        /// <remarks>
+        /// Vector axes are the following:
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </remarks>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometerWithoutCorrection(bool waitForData = true) => ReadMagnetometerWithoutCorrection(waitForData, DefaultTimeout);
+
+        /// <summary>
+        /// Read the magnetometer without Bias correction and can wait for new data to be present
+        /// More info, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L921
+        /// </summary>
+        /// <remarks>
+        /// Vector axes are the following:
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </remarks>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometerWithoutCorrection(bool waitForData, TimeSpan timeout)
+        {
+            SpanByte rawData = new byte[8];
+
+            // Wait for a data to be present
+            if (waitForData)
+            {
+                DateTime dt = DateTime.UtcNow.Add(timeout);
+                while (!HasDataToRead)
+                {
+                    if (DateTime.UtcNow > dt)
+                    {
+                        throw new TimeoutException($"{nameof(ReadMagnetometerWithoutCorrection)} timeout reading value");
+                    }
+                }
+            }
+
+            ReadBytes(Register.HXL, rawData);
+
+            Vector3 magnetoRaw = new Vector3();
+
+            /* Shift the MSB data to left by 5 bits */
+            /* Multiply by 32 to get the shift left by 5 value */
+            magnetoRaw.X = (rawData[1] & 0x7F) << 5 | rawData[0] >> 3;
+            if ((rawData[1] & 0x80) == 0x80)
+            {
+                magnetoRaw.X = -magnetoRaw.X;
+            }
+
+            /* Shift the MSB data to left by 5 bits */
+            /* Multiply by 32 to get the shift left by 5 value */
+            magnetoRaw.Y = (rawData[3] & 0x07F) << 5 | rawData[2] >> 3;
+            if ((rawData[3] & 0x80) == 0x80)
+            {
+                magnetoRaw.Y = -magnetoRaw.Y;
+            }
+
+            /* Shift the MSB data to left by 7 bits */
+            /* Multiply by 128 to get the shift left by 7 value */
+            magnetoRaw.Z = (rawData[5] & 0x07F) << 7 | rawData[4] >> 1;
+
```

Comment:
additional line not needed to be consistent with the previous lines

Reply:
fixed